### ISSUE TITLE
gcc 4.8 too old

### ISSUE THREAD

**Issue by matt-chan** [theochem/cgbasis#19]
@tovrstra So we ran into this problem earlier but I never noticed it. The compile fails because the GCC version is too old I think. I'm actually not sure how it even managed to succeed in builds like [this](https://travis-ci.org/theochem/gbasis/jobs/278420508). 

There's gcc 7.2.0 in the [anaconda channel](https://anaconda.org/anaconda/repo), but it looks like the compiler is only available on Linux. Libgcc (of various versions) is available on all systems though the anaconda channel. I think the reason they don't include the osx compiler is because they want them to use clang (which is also available in the anaconda channel). 

So I guess we're out of options? We should be building our codes with modern C++ requirements on clang for osx and gcc for linux. I don't even know if libint/libxc build successfully in clang. Another possibility is that we use the gcc that's included in Travis, and include the anaconda libgcc for the corresponding version? That way we don't have to rebuild our _entire_ toolchain... (technically though, everything we install from conda is already using clang to build, so we're actually doing something bad at the moment...)

**Comment by matt-chan** [comment]
Investigated a bit. 
Libint should work on clang. Libxc looks like it might (homebrew seems to use it).

So should we repackage the whole thing on OSX with clang then?

**Comment by matt-chan** [comment]
A bit more: 
- We can use gcc7 from travis. This isn't a problem I think.
- We can't use gcc7 on OSX. There's no libgcc > 5.2 (or even 4.8.5?) on OSX, even in the anaconda channel.

So we have to rebuild everything using clang on OSX.... Sigh. I'm working on that now.

**Comment by matt-chan** [comment]
Looks like the real issue comes from the new version of conda? Python-Cython-ci-demo is also breaking. They changed to their own version of ld, which forces us to use the conda compilers. This is messy!

https://github.com/conda/conda/issues/6030#issuecomment-333331285

**Comment by matt-chan** [comment]
Okay. Fixed the build I think.

It looks like anaconda moved their python versions to build against libgcc7.2, so we had to use compiler("c") to get the right version of gcc.

The build now uses gcc 7.2 on linux, and (I think) clang on osx. I'll be able to see once the (ridiculous) 200 job OSX backlog clears.... 

We should be able to use this to build libint and libxc in the future. I'm not sure if we want to use travis infrastructure or our own. I don't like the idea of having to find an OSX machine locally. Conda technically can cross-compile, but you have to install a version of the OSX SDK somehow on your linux box.

**Comment by matt-chan** [comment]
@tovrstra Okay. _Finally_ fixed everything. 

CXXFLAGS is ignored in Cython/Setuptools so I had to add extra code to setup.py to extract it. I also filed a bug report. Hopefully it'll become standard. https://github.com/pypa/setuptools/issues/1192

So to summarize: 
- Use compiler("c")/compiler("cxx") in meta.yml instead of gcc/libgcc
- Add the additional code to setup.py from #21 to get the CXXFLAGS so OSX compiles.
